addons/onboarding/models/onboarding_onboarding.py

```diff
@@ -0,0 +1,84 @@
+    _name = 'onboarding.onboarding'
+    _description = 'Onboarding'
+
+    name = fields.Char('Name of the onboarding', translate=True)
+    alias = fields.Char('One word alias', translate=False)
+    step_ids = fields.One2many('onboarding.onboarding.step', 'onboarding_id', 'Onboarding steps')
+
+    panel_background_color = fields.Selection([('orange', 'Orange'), ('blue', 'Blue')], default='orange')
+    panel_background_image = fields.Binary("Panel's background image")
+    panel_close_action_name = fields.Char('Action to execute when closing the panel')
+
+    is_per_company = fields.Boolean('Should be done per company?', default=True)
+
+    current_progress_id = fields.One2many(
+        'onboarding.progress', 'onboarding_id', 'Onboarding Progress Record', compute='_compute_progress_id')
+
+    progress_ids = fields.One2many('onboarding.progress', 'onboarding_id', 'Onboarding Progress Records')
+
+    _sql_constraints = [
+        ('name_uniq', 'UNIQUE (name)', 'Onboarding name must be unique.'),
+        ('alias_uniq', 'UNIQUE (alias)', 'Onboarding alias must be unique.'),
+    ]
+
+    @api.depends_context('company')
+    @api.depends('progress_ids')
+    def _compute_progress_id(self):
+        per_company = self.filtered_domain([('is_per_company', '=', True)])
+        existing_progress = self.env['onboarding.progress'].search([
+            '|',
+            ('onboarding_id', 'in', (self - per_company).ids),
+            '&', ('onboarding_id', 'in', per_company.ids),
+                 ('company_id', '=', self.env.company.id),
+        ])
+
+        for onboarding in self:
+            if onboarding in existing_progress.onboarding_id:
+                onboarding.current_progress_id = existing_progress.filtered(
+                    lambda progress: (progress.onboarding_id == onboarding
+                                      and (progress.company_id.id in {False, self.env.company.id})))
+            else:
+                onboarding.current_progress_id = False
+
+    def close(self):
+        """Close the onboarding panel."""
+        self.current_progress_id.close()
+
+    def write(self, values):
+        if 'is_per_company' in values:
+            onboardings_per_company_update = self.filtered(
+                lambda onboarding: onboarding.is_per_company != values['is_per_company'])
+
+        res = super().write(values)
+
+        if 'is_per_company' in values:
+            # When changing this parameter, all progress (onboarding and steps) is reset.
+            onboardings_per_company_update.progress_ids.unlink()
+        return res
+
+    def search_or_create_progress(self):
+        """Create Progress + Progress Steps records as necessary for the context.
+        """
+        onboardings_without_progress = self.filtered(lambda onboarding: not onboarding.current_progress_id)
+        onboardings_without_progress.create_progress_records()
+        onboardings_without_progress.step_ids.ensure_progress_exist()
```

Comment:
Pushing a bit here, but would it be possible to have the same logic for steps?
aka created by the controller when accessed, that way we avoid pre-creating stuff?